posts/2015-12-29-an-angular-developers-first-react-app.md

```diff
@@ -0,0 +1,187 @@
+
+#### But what about separation of concerns? This just feels wrong.
+
+I totally understand your trepidation, but which concerns were we actually separating with MVC? In Angular, you had to remember to inject the `$scope` service into your controller. Then, you had to assign your models to `$scope` variables so that you could access them in your view. We're exerting extra effort to pass information from our controller to our view. In MVC, the view can do very little without help from the controller. The intent behind *Separation of Concerns*, in my opinion, is to encourage the use of small single-purpose components. Including your UI markup inside your component class using JSX doesn't violate this principle, but it *does* make things a little easier for the developer.
+
+In React, the paradigm is slightly different from Angular. Each React component knows how to handle itself, but *only* itself. It knows how to take input and render its own UI. Putting the rendering code inside of the component class really makes sense. If you don't agree immediately, I get it. But, just try it out on something tangible (not a demo or tutorial) and see how you feel about it. If you still don't like it, JSX is not a requirement. You can build your React components without it.
+
+Let's get on with this walk through.
+
+
+## Let's Set Some Attributes... er... eh... Properties
+
+Once you get over the weird (but strangely familiar) JSX syntax, it is pretty easy to see the correlation between a React component and an Angular directive. The next logical step is to get some data into our new component. In Angular, I often create directives with isloate scopes. To pass values into the directive you use attributes. In React, the concept is similar, but the syntax is a little different. Let's take a look:
+
+```javascript
+var Title = React.createClass({
+  render: function() {
+    return (
+      <h1>{this.props.title}</h1> 
+    ); 
+  } 
+});
+```
+[Don't forget to follow along in the JSFiddle](http://jsfiddle.net/michaelishere/52oxac3k/)
+
+That's it for the component definition. React components have a `props` property with key/value pairs for all attributes passed to them through attributes in JSX. To use this component, we just need to supply the value of the `title` property like this:
+
+```javascript
+<Title title={myTitle} />
+```
+
+Notice that in JSX, you don't need any single or double quotations surrounding your property value. You just enclose the `myTitle` variable with a single set of curly braces. Hopefully, this feels a lot like Angular's double curly brace syntax for binding to `$scope` variables. Just remember, no quotes.
+
+
+#### Statless Immutable Glory
```

Comment:
think you could proof read and fix up a few of these little mistakes? Then I'll merge it for you :)